lib/internal/errors.js

```diff
@@ -372,6 +356,23 @@ function makeSystemErrorWithCode(key) {
   };
 }
 
+function makeNodeErrorForHideStackFrame(Base) {
+  class HideStackFramesError extends Base {
```

Comment:
Won't this mess with the `.name of the error and how stack traces look (vs the previous implementation)?

Reply:
The only reason i created this class, was because to avoid to patch everywhere the logic for Error.stackTraceLimit over and over.

The name should be the same imho. But i can tackle it.

Reply:
Thanks let me know when ready